### PR TITLE
Allow passing strings to the pay buttons by simplifying the configuration of the main component

### PR DESCRIPTION
**Pull request by JlUgia**
This PR simplifies the logic internally to enable loading payment configurations from multiple formats. In addition, it:
* Uses a map to limit one configuration per payment provider
* Uses enums instead of strings to define configuration pairs
* Removes text file serialization logic from the core payment client

**NOTE:** This change stems from #7 and @adxzhang1's [pull request](https://github.com/google-pay/flutter-plugin/pull/50). This PR proposes the removal of using one parameters per format (asset file name or JSON string) in the `PayButton` component. Instead, it simplifies the component, allowing the configuration to be passed only as a `PaymentConfiguration` object. This object contains methods to create a `PaymentConfiguration` object from multiple formats (`fromJSONString`, `fromAsset`), hence the responsibility to load the configuration is shifted to the implementer, resulting in simpler business logic in the package without making the integration more complex for developers (with the exception of loading the configuration from asset, which requires an additional 2-liner).

### CODE REVIEW

**Comment by JlUgia** [comment]
@adxzhang1 Check out this proposal to allow passing payment configuration as strings. It's a slight variation from the initial proposal, which keeps the core of the library simpler and focused on its main purpose without adding complexity to the surfacing layer.
It'd be great to hear your thoughts.

**Comment by JlUgia** [comment]
@domesticmouse added you as an optional in case you can take a look (it's a small one)

File: pay/lib/src/pay.dart [hunk 1]
```diff
@@ -24,44 +24,21 @@ class Pay {
   final PayPlatform _payPlatform;
 
   /// List of configurations for the payment providers targeted.
```
Comment (niccorder) [inline pay/lib/src/pay.dart:26]:
```suggestion
  /// Map of configurations for the payment providers targeted.
```

File: pay/lib/src/pay.dart [hunk 2]
```diff
@@ -72,12 +49,21 @@ class Pay {
...
+    throwIfProviderIsNotDefined(provider);
+    return _payPlatform.showPaymentSelector(
+        _configurations[provider]!, paymentItems);
+  }
+
+  /// Verifies that the selected provider has been previously configured or
+  /// throws otherwise.
+  void throwIfProviderIsNotDefined(PayProvider provider) {
+    if (!_configurations.containsKey(provider)) {
+      throw Exception(
```
Comment (niccorder) [inline pay/lib/src/pay.dart:65]: **[non-blocking]** Would be nice here to have a custom throwable for this case should the consumer want to handle accordingly. (e.g. `InvalidPaymentProvider`)

**Review by niccorder — APPROVED** [review]
Approving due to non-blocking comments. 

The change definitely looks better for consumers 🎉

File: pay/lib/src/pay.dart (hunk 2 above)
Comment (JlUgia) [reply pay/lib/src/pay.dart:65]: Good idea, will add

Comment (JlUgia) [reply pay/lib/src/pay.dart]: Added in commit f52026e.
PTAL @niccorder

Comment (niccorder) [reply pay/lib/src/pay.dart]: Looks great – thanks Jose!

**Comment by JlUgia** [comment]
Addresses #7

**Comment by Nidacoke** [comment]
4d612b11f05042e204804a9c176afbe8f6ed4a3b

**Pull request merged**